prow/plank/plank.go

```diff
@@ -27,16 +27,18 @@ import (
 
 // NewProwJob initializes a ProwJob out of a ProwJobSpec.
 func NewProwJob(spec kube.ProwJobSpec) kube.ProwJob {
+	jobUUID := uuid.NewV1().String()
 	return kube.ProwJob{
 		APIVersion: "prow.k8s.io/v1",
 		Kind:       "ProwJob",
 		Metadata: kube.ObjectMeta{
-			Name: uuid.NewV1().String(),
+			Name: jobUUID,
 		},
 		Spec: spec,
 		Status: kube.ProwJobStatus{
 			StartTime: time.Now(),
 			State:     kube.TriggeredState,
+			UUID:      jobUUID,
```

Comment:
My only reason behind going with Name vs UID: Is there any case where I have the location of the bucket, but I want to inspect the ProwJob manifest? Or do we assume that all info that can be carried as part of the manifest, will end up being in the bucket? I can't think of something atm but using the Name instead of something else will allow us to Get the PJ directly instead of having to List all prowjobs and filter out based on UID (assuming the PJ is not garbage-collected).